src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -124,21 +130,34 @@ def get_responses(self) -> pd.DataFrame:
             "sex",
             "race_ethnicity",
             "census_year",
+            "guardian_1",
+            "guardian_1_address",
+            "guardian_2",
+            "guardian_2_address",
+            "housing_type",
         ])
 
     def to_observe(self, event: Event) -> bool:
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April  FIXME: cooler to do this with the timestep
-                return True
-        return False
-
+        return ((self.clock().year % 10 == 0)  # decennial year
+                and (self.clock().month == 4)  # month of April
+                and (self.clock().day <= self.time_step)  # time step containing first day of month
+                      # with current model spec, on first time step, event.time equals 2020-04-29
+                      # but self.clock() equals 2020-04-01
+               )
     def do_observation(self, event) -> None:
         pop = self.population_view.get(
-            event.index, # query="alive == 'alive'",  # TODO: uncomment this to include only living simulants in census
+            event.index,
+            query="alive == 'alive'",  # census should include only living simulants
         )
         pop["middle_initial"] = pop["middle_name"].astype(str).str[0]
         pop = pop.drop(columns="middle_name")
 
-        # TODO: include additional columns specified in MIC-3642
+        # merge addresses for guardian_1 and guardian_2 for the rows with guardians
+        for i in [1,2]:
+            s_guardian_id = pop[f"guardian_{i}"].dropna()
+            s_guardian_id = s_guardian_id[s_guardian_id != -1] # is it faster to remove the negative values?
```

Comment:
I don't know if it's faster but it might be safer in case future work uses other negative values for special cases